### PR TITLE
enh: add compatibility with zfit 0.6

### PR DESCRIPTION
**Pull request by mayou36**
In order to allow a loss to have optimizations in there and for an easier creation of a new loss, added a method that creates a new loss and therefore reduces the reliability on a specific implementation when creating a new loss.

The new method is `create_new`. Added this in the implementation and the interface check

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -6,11 +6,11 @@ dependencies:
...
-  - tensorflow>=2.0
+  - tensorflow>=2.4
   - tensorflow-probability
   #- zfit
   - asdf
   - matplotlib
   - pip:
     - .
-    - zfit
+    - zfit =>0.6.4
```
Comment (eduardo-rodrigues) [inline environment.yml:16]: Do  you not mean >= instead of => ?

File: src/hepstats/hypotests/hypotests_object.py [hunk 2]
```diff
@@ -157,8 +158,18 @@ def lossbuilder(self, model, data, weights=None):
...
-        loss = type(self.loss)(model=model, data=data)
-        loss.add_constraints(self.constraints)
+        if hasattr(self.loss, "create_new"):
+            loss = self.loss.create_new(
+                model=model, data=data, constraints=self.constraints
+            )
+        else:
+            warnings.warn(
+                "A loss should have a `create_new` method. If you are using zfit, please make sure to"
+                "upgrade to >= 0.6.3",
```
Comment (eduardo-rodrigues) [inline src/hepstats/hypotests/hypotests_object.py:168]: You mention 0.6.4 above.

File: environment.yml (hunk 1 above)
Comment (mayou36) [reply environment.yml]: Indeed, typo, fixed

File: src/hepstats/hypotests/hypotests_object.py (hunk 2 above)
Comment (mayou36) [reply src/hepstats/hypotests/hypotests_object.py]: Indeed, fixed

**Comment by mayou36** [comment]
docs and format CI fail for other reasons, they alreadf failed before. I'll address that in a different PR.
Fine to merge?

**Review by eduardo-rodrigues — APPROVED** [review]

**Comment by eduardo-rodrigues** [comment]
Fine by me :-). Up to you to wait or not for @marinang.

There is only a little issue with formatting/CI?

**Review by marinang — APPROVED** [review]
